### ISSUE TITLE
Enable/Disable Endpoint should show timestamp

### ISSUE THREAD

**Issue by WtfJoke** [codecentric/chaos-monkey-spring-boot#246]
When doing gamedays we usually write down the exact moment we started the experiment.
Usually we enable/disable chaos monkey with the endpoints `/chaosmonkey/enable` and `/chaosmonkey/disable`. The response could extended with a human readable timestamp.

So the response has a small benefit.

**Comment by MatthewBerk** [comment]
Hello @WtfJoke !
I am new to open-source contributions and I would like to attempt taking on this issue/feature adding. If you have any tips or recommendations for where to start, that would be greatly appreciated.  Thank you!

I will be attempting to load this project on my system later today!

**Comment by WtfJoke** [comment]
Hi @MatthewBerk 

Sure go ahead and start working on it.  Appreciate it that you want to contribute to open source 😃 

The best start would be [ChaosMonkeyRestEndpoint](https://github.com/codecentric/chaos-monkey-spring-boot/blob/main/chaos-monkey-spring-boot/src/main/java/de/codecentric/spring/boot/chaos/monkey/endpoints/ChaosMonkeyRestEndpoint.java) and [ChaosMonkeyJmxEndpoint](https://github.com/codecentric/chaos-monkey-spring-boot/blob/main/chaos-monkey-spring-boot/src/main/java/de/codecentric/spring/boot/chaos/monkey/endpoints/ChaosMonkeyJmxEndpoint.java). In these two classes is the logic for enabling/disabling Chaos Monkey. This should be a good starting point.

**Comment by MatthewBerk** [comment]
Thank you for assigning me and for the advice. 😃

It may take me a while since I will be working on gaining an understanding of the project while trying to add this improvement. I hope that is fine with you.

**Comment by WtfJoke** [comment]
No worries, take your time. :) If you have questions feel free to ask.

**Comment by MatthewBerk** [comment]
[I didn't want to raise a new issue in case this isn't that important]
I wanted to note a small hiccup I had when trying to run this project. So when I fork and pull the project into my setup, (whether I open it in IntlelliJ first or not) and execute ./mvnw clean verify , I get this error
![image](https://user-images.githubusercontent.com/51866904/126854150-49d01b4b-461e-4b37-8f55-01784ab52cce.png)

This solution I found online fixed it, https://stackoverflow.com/a/52750359
![image](https://user-images.githubusercontent.com/51866904/126854175-4229064e-9538-40a7-a34c-b0867960a5e2.png)

Though it lead to this
![image](https://user-images.githubusercontent.com/51866904/126854189-e59e0905-a8c2-41a9-a694-17566ed9b786.png)
Files getting modified that have not been updated on main repo in 2 years!

I am still a novice when it comes to Maven. Is this fine or should I be running a different command?

**Comment by WtfJoke** [comment]
Hmm strange enough, I never experienced this error allthough I set up the repo on multiple new machines recently. 

It seems like the command `./mvnw -N io.takari:maven:wrapper` should have executed after the version of the maven wrapper was updated (see [here](https://github.com/codecentric/chaos-monkey-spring-boot/blob/main/.mvn/wrapper/maven-wrapper.properties)). But the diff doesnt look like there seems to be much changes. 

Long story short, please open a PR for this change in these three files.  :) 

Out of curiosity which java version do you use?

**Comment by MatthewBerk** [comment]
Done! 
I made sure it still worked before I pushed
![image](https://user-images.githubusercontent.com/51866904/126881253-d8eaea83-2da9-437e-8efa-e64fede38a74.png)

Originally I was using Java 1.8, though I switched to Java 11.0.9 and experienced the same issue from fresh pull!

**Comment by MatthewBerk** [comment]
I found that the `enableChaosMonkey(),` `disableChaosMonkey()` methods in  [ChaosMonkeyRestEndpoint](https://github.com/codecentric/chaos-monkey-spring-boot/blob/40de08a801e1882885a9ba852708896d812ae055/chaos-monkey-spring-boot/src/main/java/de/codecentric/spring/boot/chaos/monkey/endpoints/ChaosMonkeyRestEndpoint.java) and [ChaosMonkeyJmxEndpoint ](https://github.com/codecentric/chaos-monkey-spring-boot/blob/40de08a801e1882885a9ba852708896d812ae055/chaos-monkey-spring-boot/src/main/java/de/codecentric/spring/boot/chaos/monkey/endpoints/ChaosMonkeyJmxEndpoint.java) only seem to execute when I run ./mvnw clean verify. They don’t get executed when I run any of the 5 Chaos Monkey demo app’s.

Is there an available demo that enables/disables Chaos Monkey so I can ensure the the timestamps are properly displayed?


Would this timestamp format be suitable or did you have something else in mind?
![image](https://user-images.githubusercontent.com/51866904/126884329-5a37b1b2-33eb-4c46-a5b7-28e88acab063.png)

**Comment by WtfJoke** [comment]
They get executed if you make a POST request to `/actuator/chaosmonkey/enable` (/disable).

Therefore start a demo app (for example [chaos-monkey-demo-app](https://github.com/codecentric/chaos-monkey-spring-boot/tree/main/demo-apps/chaos-monkey-demo-app)) and then make a POST request to eg. `localhost:8888/actuator/chaosmonkey/enable`. Then the method is executed.

I personally use [Insomnia](https://insomnia.rest/download) for the POST request, but you can use any rest client of your choice. If you are more of a command line person you can also use following `curl` command:

`curl --request POST --url http://localhost:8888/actuator/chaosmonkey/enable`  

> Would this timestamp format be suitable or did you have something else in mind?

The suggested timestamp format looks good to me. Not sure if the name "timestamp" is the perfect choice. Maybe "activatedOn" or "activationTimestamp" or something similar suits better. Im open for suggestions :D

**Comment by MatthewBerk** [comment]
I will look into that Insomnia program! 😃

I feel this form seems good!
ActivatedAt: "2021-07-24T23:40:26.698"
DeactivatedAt: "2021-07-24T25:20:06.301"

**Comment by MatthewBerk** [comment]
I forgot to ask this. Do you want the timestamp to reflect the users current time or stick with UTC time zone?

Launching a chaos monkey demo and going to localhost:8888 lead to this being display:
![image](https://user-images.githubusercontent.com/51866904/127426286-a7a23554-d5e8-4f22-b743-5fd9590d852c.png)
The timestamp is for UTC time zone.

**Comment by WtfJoke** [comment]
Think both formats has its down/upsides.

UTC: 
+ no problems with zone converting
- user might need to convert zone by itself

Server time zone:
+ most likekly better to read for user
-  timezone of server might differ than from the user
-  timezone conversion could lead to problems

However I think I still would preferr servers time zone. I think its suits the purpose better.

**Comment by checketts** [comment]
I second the server time zone. 

  Also, perhaps an additional attribute of 'duration' that would be human readable would help the end user understand at a glance.

**Comment by WtfJoke** [comment]
> Also, perhaps an additional attribute of 'duration' that would be human readable would help the end user understand at a glance.

Do you mean the duration of how long chaos monkey was enabled (when it gets disabled)?

**Comment by checketts** [comment]
Both if it were disable and if it is still active. Seeing something : `durationMinutes: 180` make for some simple math so the caller doesn't have to worry about when something was enabled (regardless of timezone).

Duration Is probably a poor name. `minutesActive` might be clearer. It would keep counting up if it hasn't been disabled. 

Feel free to ignore this idea though if it distracts from the endpoints purpose.

**Comment by WtfJoke** [comment]
I think its a great idea! However I would do that separately. But it would be a nice addition in my opinon. Maybe create a separate issue for that.

**Comment by MatthewBerk** [comment]
> Think both formats has its down/upsides.
> 
> UTC:
> 
> * no problems with zone converting
> 
> * user might need to convert zone by itself
> 
> Server time zone:
> 
> * most likely better to read for user
> 
> * timezone of server might differ than from the user
> * timezone conversion could lead to problems
> 
> However I think I still would preferr servers time zone. I think its suits the purpose better.

So just to confirm, you want user specific time format, not UTC? Since the UTC timezone format is what is used on the webpage when error occurs!

**Comment by MatthewBerk** [comment]
> Both if it were disable and if it is still active. Seeing something like: `durationMinutes: 180` makes for some simple math so the caller doesn't have to worry about when something was enabled (regardless of timezone).
> 
> Duration Is probably a poor name. `minutesActive` might be clearer. It would keep counting up if it hasn't been disabled.
> 
> Feel free to ignore this idea though if it distracts from the endpoint's purpose.

Would possibly converting it to hours be better? like if had 190 minutes. would do duration: 3 Hours, 10 Minutes.

**Comment by MatthewBerk** [comment]
> I think its a great idea! However I would do that separately. But it would be a nice addition in my opinon. Maybe create a separate issue for that.

I agree with @checketts . That seems useful! Would we want to include hours if minutes is large enough?

**Comment by WtfJoke** [comment]
I'll close this here (as #254 has been merged, I referenced @checketts comment in a new issue https://github.com/codecentric/chaos-monkey-spring-boot/issues/258.
